server/container/spring/src/main/java/org/apache/james/container/spring/bean/factorypostprocessor/MailboxConfigurationBeanFactoryPostProcessor.java

```diff
@@ -96,10 +112,21 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
             registry.registerAlias(mailboxIdDeserializer, "mailbox-id-deserializer");
             registry.registerAlias(mailboxIdFactory, "mailboxIdFactory");
 
+            removeMailboxManagersExceptRightSelectedOne(registry, mailbox);
+
         } catch (ConfigurationException e) {
             throw new FatalBeanException("Unable to config the mailboxmanager", e);
         }
 
     }
 
+    private void removeMailboxManagersExceptRightSelectedOne(BeanDefinitionRegistry registry, String selectedMailboxManager) {
+        mailboxManagerIds.remove(selectedMailboxManager);
```

Comment:
However you can still avoid this remove : 

```
        for (String otherMailboxManager : mailboxManagerIds) {
            if (registry.containsBeanDefinition(otherMailboxManager) && !otherMailboxManager.equals(selectedMailboxManager)) {
                registry.removeBeanDefinition(otherMailboxManager);
            }
        }
```

This would allow mailboxManagerIds to be private, static, final and immutable. All of what we like !

Of course name the condition by extracting it in a method ;-)